website/content/v1.10/talos-guides/configuration/reproducible-machine-config.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "Reproducible Machine Configuration"
+description: "How to reliably and consistently regenerate Talos machine configs from source inputs over time."
+---
+
+Talos uses declarative configuration for clusters, but upgrades can cause drift between the declared machine configuration (files you keep in Git) and the deployed configuration (what’s actually running on your nodes).
+
+To prevent this drift, we recommend discarding full declared configuration files and instead using a patch-based workflow to regenerate machine configuration whenever you need them.
+
+After an upgrade, you’ll need the following inputs to reliably regenerate configuration:
```

Comment:
Why after an upgrade? It should say something like "preserve" following inputs to perform reproducible machine configuration generation.